specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,270 @@
+  /// populate the Response property, then WebView2 will show the default
+  /// authentication challenge dialog prompt to the user.
+  ///
+  HRESULT add_BasicAuthenticationRequested(
+    [in] ICoreWebView2BasicAuthenticationRequestedEventHandler* eventHandler,
+    [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_WebResourceRequested.
+  HRESULT remove_BasicAuthenticationRequested(
+      [in] EventRegistrationToken token);
+}
+
+/// This is the CoreWebView2BasicAuthenticationRequestedEventHandler interface
+[uuid(f87e5d35-3248-406b-81dd-1c36aab8081d), object, pointer_default(unique)]
+interface ICoreWebView2BasicAuthenticationRequestedEventHandler : IUnknown
+{
+  /// Called to provide the implementer with the event args for the
+  /// corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] ICoreWebView2BasicAuthenticationRequestedEventArgs* args);
+}
+
+/// Represents a Basic HTTP authentication response that contains a user name
+/// and a password as according to RFC7617 (https://tools.ietf.org/html/rfc7617)
+[uuid(bc9cfd60-29c4-4943-a83b-d0d2f3e7df03), object, pointer_default(unique)]
+interface ICoreWebView2BasicAuthenticationResponse : IUnknown
+{
+  /// User name provided for authentication.
+  [propget] HRESULT UserName([out, retval] LPWSTR* userName);
+  /// Set user name property
+  [propput] HRESULT UserName([in] LPCWSTR userName);
+
+  /// Password provided for authentication
+  [propget] HRESULT Password([out, retval] LPWSTR* password);
+  /// Set password property
+  [propput] HRESULT Password([in] LPCWSTR password);
+}
+
+/// Event args for the BasicAuthenticationRequested event. Will contain the
+/// request that led to the HTTP authentication challenge, the challenge
+/// and allows the host to provide credentials response or cancel the request.
+[uuid(51d3adaa-159f-4e48-ad39-a86beb2c1435), object, pointer_default(unique)]
+interface ICoreWebView2BasicAuthenticationRequestedEventArgs : IUnknown
+{
+  /// The web resource request that led to the authentication challenge
+  [propget] HRESULT Request([out, retval] ICoreWebView2WebResourceRequest** request);
+
+  /// The authentication challenge string
+  [propget] HRESULT Challenge([out, retval] LPWSTR* challenge);
+
+  /// Response to the authentication request with credentials. This object will be populated by the app
+  /// if the host would like to provide authentication credentials.
+  [propget] HRESULT Response([out, retval] ICoreWebView2BasicAuthenticationResponse** response);
+  
+  /// Cancel the authentication request. False by default.
+  /// If set to true, Response set will be ignored.
+  [propget] HRESULT Cancel([out, retval] BOOL* cancel);
+  /// Set the Cancel property.
+  [propput] HRESULT Cancel([in] BOOL cancel);
+
+  /// Returns an `ICoreWebView2Deferral` object.  Use this operation to
+  /// complete the event at a later time.
```

Comment:
Would be good to explicitly say here what is getting blocked (basic auth dialog (depending on cancel) and sending the next request with auth info (depending on cancel/response))